queries.sql

```diff
@@ -74,4 +74,25 @@ SELECT animals.name, full_name, species.name FROM animals JOIN owners on owners.
 /* Question 6 */
 SELECT name, full_name, escape_attempts FROM animals JOIN owners on owners.id = animals.owner_id WHERE full_name = 'Dean Winchester' and escape_attempts = 0;
 /* Question 7 */
-SELECT full_name, COUNT(*) as total  FROM animals JOIN owners on owners.id= animals.owner_id GROUP BY full_name ORDER BY total DESC LIMIT 1;
\ No newline at end of file
+SELECT full_name, COUNT(*) as total  FROM animals JOIN owners on owners.id= animals.owner_id GROUP BY full_name ORDER BY total DESC LIMIT 1;
+
+/* Part 4 */
+
+/* Question 1 */
+SELECT vets.name, date_of_visit, animals.name FROM visits JOIN animals on animals.id = visits.animals_id JOIN vets on vets.id = visits.vets_id WHERE vets.name = 'William Tatcher' ORDER BY date_of_visit DESC LIMIT 1;
+/* Question 2 */
+SELECT vets.name, species.name, COUNT(species.name) as total_animals FROM visits JOIN animals on animals.id = visits.animals_id JOIN species on species.id = animals.species_id  JOIN vets on vets.id = visits.vets_id WHERE vets.name = 'Stephanie Mendez' GROUP BY species.name, vets.name;
+/* Question 3 */
+SELECT vets.name, species.name as specialization FROM specializations FULL JOIN vets on vets.id = specializations.vets_id FULL JOIN species ON species.id = specializations.species_id;
```

Comment:
Question 3: 

```sql
SELECT vets.name, species.name as specialization FROM specializations FULL JOIN vets on vets.id = specializations.vets_id FULL JOIN species ON species.id = specializations.species_id;
```

This query will not work because the FULL JOIN keyword is not supported by all SQL dialects. You can use the INNER JOIN keyword instead, which will only return rows where the vets and species tables have a matching record.

Here is one way of writing the query for the question

```sql
SELECT vets.name, species.name AS specialization
FROM vets
INNER JOIN specializations ON vets.id = specializations.vets_id
INNER JOIN species ON species.id = specializations.species_id;

```

Reply:
The question is to include all not just the intersection. Inner join just present intersection.